### PR TITLE
Jest mocks

### LINKED ISSUE #3788: Remove 'sinon' dependencies in favor of jest mocks
Use the jest mocking functions instead of sinon.

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Replace sinon with jest mockers
1. Remove unused dev dependencies
1. Remove unused npm scripts
1. Rewrite frontend_build tests for jest
1. Fix warnings comiong out of tests

### Reviewer guidance

…

### References

Resolves #3788 

…

----

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -15,6 +15,9 @@ module.exports = {
         node: true,
         jest: true,
       },
+      parserOptions: {
+        ecmaVersion: 8,
```
Comment (rtibbles) [inline .eslintrc.js:19]: :+1:

**Review by rtibbles — CHANGES_REQUESTED** [review]
Great cleanup. Just need to add the frontend_build/src files to collectCoverageFrom in the jest.conf.js, and it will be a seamless refactor!

**Comment by jonboiser** [comment]
@rtibbles added the frontend_build directory to the coverage report

**Comment by rtibbles** [comment]
Buildkite failure is unrelated.

**Review by rtibbles — APPROVED** [review]

**Comment by rtibbles** [comment]
Merged as buildkite failure is unrelated.

**Pull request merged**